src/components/common/dashboard/OperationComponent.vue

```diff
@@ -0,0 +1,388 @@
+                        operation.value.missingApprovals.groups.includes(role.value)) &&
+                    !isRejected
+                );
+            });
+
+            const sentApprove = ref(store.getters.processedOperations.approved.includes(operation.value.operationId));
+            const sentReject = ref(store.getters.processedOperations.rejected.includes(operation.value.operationId));
+            const provideReason = ref(false);
+            const selectedReason = ref("");
+            const finalReason = ref("");
+            const isMyOperation = operation.value.initiator == props.enrollmentId;
+            const showWatchOption = !isMyOperation && isPending;
+
+            const username = ref("...");
+            const isAdmin = computed(() => {
+                return role.value == Role.ADMIN;
+            });
+
+            const isWatched = ref(props.watched);
+
+            const dateFormatOptions = {
+                weekday: "short",
+                year: "numeric",
+                month: "numeric",
+                day: "numeric",
+                hour: "2-digit",
+                minute: "2-digit",
+            };
+
+            const params = ref([] as string[]);
+            const title = ref("");
+            const shownOpId = operation.value.operationId.substring(0, 4);
+
+            const initiatedTimestamp = new Date(operation.value.initiatedTimestamp).toLocaleString("en-US", dateFormatOptions);
+            const lastUpdateTimestamp = new Date(operation.value.lastModifiedTimestamp).toLocaleString("en-US", dateFormatOptions);
+
+            const showDetails = ref(false);
+
+            onBeforeMount(async () => {
+                await getRole();
+                if (isAdmin.value) {
+                    await getNameByEnrollmentId();
+                }
+                await createDisplayObjects();
+            });
+
+            const type = getOperationBadgeIdentifier(operation.value);
+
+            const statusColor = computed(() => {
+                switch ((props.operation as Operation).state) {
+                    case OperationStatus.PENDING:
+                        return "bg-blue-300";
+                    case OperationStatus.FINISHED:
+                        return "bg-green-300";
+                    case OperationStatus.REJECTED:
+                        return "bg-red-300";
+                }
+            });
+
+            watch(selectedReason, () => {
+                finalReason.value = selectedReason.value;
+                if (finalReason.value == RejectionReasons.OTHER) {
+                    finalReason.value = "";
+                }
+            });
+
+            async function getRole() {
+                role.value = await store.getters.role;
+            }
+
+            async function approve() {
+                if (await approveMatriculation(operation.value)) {
+                    //TODO Set watch
```

Comment:
Remove TODO; either implement it or open a new issue and implement it later (if still needed).
Same for other TODOs in this file

Reply:
The TODOs are meant to remain in this case. See note and comments in PR https://github.com/upb-uc4/ui-web/pull/785. The functionality is currently under development on branch `feature/operations_update`. I will remove them, as I already started implementing the TODOs before this releases can be merged